subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/JdkIllegalReflectionTestWorkerIntegrationTest.groovy

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.workers.internal
+
+import org.gradle.api.JavaVersion
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.AvailableJavaHomes
+import spock.lang.Issue
+
+/**
+ * Ensures test behavior and actual application behavior are equivalent when
+ * production code attempts to perform reflection on JDK internals.
+ */
+class JdkIllegalReflectionTestWorkerIntegrationTest extends AbstractIntegrationSpec {
+
+    def setup() {
+        buildFile << """
+            plugins {
+                id 'application'
+            }
+
+            ${mavenCentralRepository()}
+
+            dependencies {
+                testImplementation 'junit:junit:4.13'
+            }
```

Comment:
If you want to avoid the explicit version number here, you can swap this out with:
```
testing.suites.test.useJUnit()
```